compiler/src/dotty/tools/backend/jvm/BCodeHelpers.scala

```diff
@@ -607,7 +607,8 @@ trait BCodeHelpers extends BCodeIdiomatic with BytecodeWriters {
       val names = tp.memberNames(takeAllFilter).toSeq.sorted
       val buffer = mutable.ListBuffer[Symbol]()
       names.foreach { name =>
-        buffer ++= tp.memberBasedOnFlags(name, required, excluded)
+        // lookup members at erasure: lampepfl/dotty#12753
+        buffer ++= atPhase(erasurePhase)(tp.memberBasedOnFlags(name, required, excluded))
```

Comment:
The change breaks forwarders for inherited trait fields:

```scala
trait T { val x = 1 }
object O extends T
```

The accessor for `x` is found and included, but it's abstract, which makes `addForwarders` skip it.

Which phase generates the field in `D$` and its concrete accessors, mixin? In Scala 2 this is done by the "fields" phase which runs before erasure.